### PR TITLE
Using new Function run endpoint

### PR DESCRIPTION
**Pull request by DeirdreCleary**
# Why

Moving to the new endpoint will give us better error messages and clearer responses generally

# How

Moved endpoint to api.evervault.com/functions/FUNCTION_NAME/runs
Also took the opportunity to deprecate encryptAndRun, listing Functions, async Function run requests, and specifying version of Function to run.

### CODE REVIEW

File: evervault/http/request.py [hunk 1]
```diff
@@ -16,7 +17,11 @@
 
 
 class Request(object):
-    auth_header_paths = ["/decrypt", "/client-side-tokens"]
+    auth_header_regexes = [
```
Comment (jakekgrog) [inline evervault/http/request.py:20]: nice!

File: evervault/http/request.py [hunk 2]
```diff
@@ -100,7 +114,7 @@ def __execute_request(self, request_object, method, url, req_params):
             **req_params,
         )
 
-    def __parse_body(self, resp, should_parse=True):
+    def __parse_body(self, resp, should_parse=True, should_raise_error_on_failure=True):
```
Comment (jakekgrog) [inline evervault/http/request.py:117]: Is this parameter doing anything?

**Review by jakekgrog — COMMENTED** [review]
LGTM, just a small nit

File: evervault/http/request.py (hunk 2 above)
Comment (DeirdreCleary) [reply evervault/http/request.py]: Nope! Thanks for catching

File: evervault/errors/error_handler.py [hunk 3]
```diff
@@ -1,7 +1,40 @@
...
+    )
+
+
+def raise_errors_on_api_error(resp, response_body):
+    if resp.status_code >= 200 and resp.status_code < 300:
+        return
+    code = response_body.get("code")
+    detail = response_body.get("detail")
+
+    if code == "unauthorized":
```
Comment (jeremywagemans) [inline evervault/errors/error_handler.py:22]: Not sure there is any benefit on introducing that many errors for non retriable errors. I'd just keep `FunctionNotReadyError` & `FunctionTimeoutError` and default on `EvervaultError` for the rest

File: .changeset/new-pugs-smoke.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"evervault-python": major
+---
+
+Migrated Function run requests to new API. Added more comprehensive error handling for Function runs. Deprecated encryptAndRun, listing Functions, async Function run requests, and specifying version of Function to run.
```
Comment (jeremywagemans) [inline .changeset/new-pugs-smoke.md:5]: I would be slightly more specific here and explain how error handling was updated. The changelog should inform users on how their integration could be impacted for breaking changes!

File: evervault/errors/error_handler.py (hunk 3 above)
Comment (DeirdreCleary) [reply evervault/errors/error_handler.py:22]: These aren't errors I introduced, they exist already in the SDK. I agree that they're not especially helpful for users, but it seemed odd for the error to exist but not used universally. Is the move here to remove its use everywhere in the SDK?

Comment (DeirdreCleary) [reply evervault/errors/error_handler.py:22]: Per offline, these changes will be made in a subsequent PR

File: .changeset/new-pugs-smoke.md (hunk 4 above)
Comment (DeirdreCleary) [reply .changeset/new-pugs-smoke.md:5]: I'll update this once we have the errors finalized in the subsequent PR 👍

**Pull request merged**